AWS Amazon API Gateway doc (#3718)
SumoLogic/sumologic-documentation: docs/integrations/amazon-aws/api-gateway.md

---
id: api-gateway
title: AWS API Gateway
description: Amazon API Gateway service allows you to create RESTful APIs, HTTP APIs, and WebSocket APIs for real-time two-way communication applications in containerized and serverless environments, as well as web applications.
---

import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/integrations/amazon-aws/AWS_API_Gateway.png')} alt="Thumbnail icon" width="50"/>

AWS API Gateway service allows you to create RESTful APIs, HTTP APIs, and WebSocket APIs for real-time two-way communication applications in containerized and serverless environments, as well as web applications.

The Sumo Logic AWS API Gateway app provides insights into API Gateway tasks while accepting and processing concurrent API calls throughout your infrastructure, including traffic management, CORS support, authorization and access control, throttling, monitoring, and API version management.

## Log and metrics types  

The AWS API Gateway app uses the following logs and metrics:

> * Amazon API Gateway metrics

Reviewer: ```suggestion
* Amazon API Gateway metrics:
```

   * [REST APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-metrics-and-dimensions.html) <img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>
   * [HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-metrics.html) <img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>
   * [WebSocket APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-websocket-api-logging.html) <img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>
* [CloudTrail API Gateway Data Event](https://docs.aws.amazon.com/apigateway/latest/developerguide/cloudtrail.html) <br/><img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>

> * API Gateway Access logs

Reviewer: ```suggestion
* API Gateway Access logs:
```

---
id: api-gateway
title: AWS API Gateway

> description: Amazon API Gateway service allows you to create RESTful APIs, HTTP APIs and WebSocket APIs for real-time two-way communication applications in containerized and serverless environments, as well as web applications.

Reviewer 2: ```suggestion
description: Amazon API Gateway service allows you to create RESTful APIs, HTTP APIs, and WebSocket APIs for real-time two-way communication applications in containerized and serverless environments, as well as web applications.
```

> * [Amazon API Gateway metrics](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-metrics-and-dimensions.html) <img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>

Reviewer 3: we may have to add separate links for http websocket and rest api metrics

* [CloudTrail API Gateway Data Event](https://docs.aws.amazon.com/apigateway/latest/developerguide/cloudtrail.html) <br/><img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>

> * [AWS Kinesis Firehose Access Log](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-logging-to-kinesis.html) <img src='https://upload.wikimedia.org/wikipedia/commons/a/a4/OOjs_UI_icon_external-link-ltr-progressive.svg' alt="Thumbnail icon" width="12"/>

Reviewer 3: Kinesis firehose access log is not log type we should say API Gateway Access log and give links to access logs for http, websocket and rest

## Collecting Logs and Metrics for AWS API Gateway

### Field in Field Schema

> Login to Sumo Logic, go to **Manage Data** > **Logs** > **Fields**. Search for the **apiname** field. If not present, create it. Learn how to create and manage fields [here](/docs/manage/fields.md#manage-fields).

Reviewer 4: ```suggestion
Log in to Sumo Logic, then go to **Manage Data** > **Logs** > **Fields**. Search for the **apiname** field. If not present, create it. Learn how to create and manage fields [here](/docs/manage/fields.md#manage-fields).
```

> ## Collecting Logs and Metrics for AWS API Gateway
> 
> ### Field in Field Schema
> 
> Log in to Sumo Logic, then go to **Manage Data** > **Logs** > **Fields**. Search for the **apiname** field. If not present, create it. Learn how to create and manage fields [here](/docs/manage/fields.md#manage-fields).

Reviewer: ```suggestion
## Collecting logs and metrics for AWS API Gateway

### Fields in field schema

Log in to Sumo Logic, then go to **Manage Data** > **Logs** > **Fields**. Search for the **apiname** field. If not present, create it. To learn how to create and manage fields, see [Fields](/docs/manage/fields.md#manage-fields).
```

### Field Extraction Rule(s)

> Learn how to create Field Extraction Rule [here](/docs/manage/field-extractions/create-field-extraction-rule).

Reviewer 4: ```suggestion
Learn how to create Field Extraction Rules [here](/docs/manage/field-extractions/create-field-extraction-rule).
```

> ### Field Extraction Rule(s)
> 
> Learn how to create Field Extraction Rules [here](/docs/manage/field-extractions/create-field-extraction-rule).
> 
> Create Field Extraction Rule for CloudTrail Logs.

Reviewer: ```suggestion
### Field extraction rules

To learn how to create field extraction rules, [Create a Field Extraction Rules](/docs/manage/field-extractions/create-field-extraction-rule).

Create a field extraction rule for CloudTrail logs:
```

```sql
Rule Name: AwsObservabilityApiGatewayCloudTrailLogsFER
Applied at: Ingest Time
Scope (Specific Data):
account=* eventname eventsource "apigateway.amazonaws.com"
Parse Expression:
| json "eventSource", "awsRegion", "responseElements", "recipientAccountId" as eventSource, region, responseElements, accountid nodrop
| where eventSource = "apigateway.amazonaws.com"
| "aws/apigateway" as namespace
| json field=responseElements "name" as ApiName nodrop
| tolowercase(ApiName) as apiname
| fields region, namespace, apiname, accountid
```

> Create Field Extraction Rule for Access Logs.

Reviewer: ```suggestion
Create a field extraction rule for access logs:
```

```sql
Rule Name: AwsObservabilityApiGatewayAccessLogsFER
Applied at: Ingest Time
Scope (Specific Data):
account=* region=* apiId domainName stage requestId status
Parse Expression:
json "apiId", "domainName", "stage" as apiId, domainName, stage
| "aws/apigateway" as namespace
| apiId as apiName
| fields apiName, namespace, apiId
```

> ### Metric Rules

Reviewer: ```suggestion
### Metric rules
```

> Create the following Metric Rule for the AWS API gateway app if not already created. Learn how to create a Metrics Rule [here](/docs/metrics/metric-rules-editor#create-a-metric-rule).

Reviewer 4: ```suggestion
Create the following Metric Rule for the AWS API Gateway app, if not already created. Learn how to create a Metrics Rule [here](/docs/metrics/metric-rules-editor#create-a-metric-rule).
```

> Create the following Metric Rule for the AWS API Gateway app, if not already created. Learn how to create a Metrics Rule [here](/docs/metrics/metric-rules-editor#create-a-metric-rule).

Reviewer: ```suggestion
Create the following metrics rule for the AWS API Gateway app, if not already created. To learn how to create a metrics rule, see [Metrics Rules Editor](/docs/metrics/metric-rules-editor#create-a-metric-rule).
```

```sql
Rule name: AwsObservabilityApiGatewayApiNameEntityRule
Metric match expression: Namespace=AWS/ApiGateway apiid=*
Variable name: apiname
Tag sequence: $apiid._1
Save it
```

> ### Collect Metrics for AWS API Gateway   

Reviewer: ```suggestion
### Collect metrics for AWS API Gateway   
```

For **Metadata**, add an **account** field to the source and assign it a value that is a friendly name/alias to your AWS account from which you are collecting metrics. This name will appear in the Sumo Logic Explorer View. Metrics can be queried via the “account field”.

### Collect Access Logs for AWS API Gateway   

1. To your Hosted Collector, add an [AWS Kinesis Firehose for Logs Source](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/).
   1. **Name**. Enter a name to display the new Source.
   2. **Description**. Enter an optional description.
   3. **Enable S3 Replay**. Do not check this option.
   4. **Source Category**. Enter `aws/apigateway`.
   5. **Fields**. Add below fields in it:
      1. Add an **account** field and assign it a value that is a friendly name/alias to your AWS account from which you are collecting logs. This name will appear in the Sumo Logic Explorer View. Logs can be queried via the `account field`. 

>       2. Add **region**, **namespace**, and **accountid** fields and assign it's respective values.

Reviewer 3: we cannot tag region and namespace need to create FER as first step.

   6. **Enable Timestamp Parsing**. Select the **Extract timestamp information from log file entries** check box.
   7. **Time Zone**. Select **Use time zone from log file. If none is detected use**, and select **Use Collector Default** from the dropdown.
   8. **Timestamp Format.** Select **Automatically detect the format**.
   9. **Enable Multiline Processing**. Select the **Detect messages spanning multiple lines** check box, and select **Infer Boundaries**.
   10. Click **Save**.

> 1. [Create Stack](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-create-stack.html) in AWS console with given [CloudFormation Template]((https://help.sumologic.com/files/KinesisFirehoseCWLogs.template.yaml)).

Reviewer 3: lets just redirect customer to this section https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/#cloudformation-template otherwise we have to maintain duplicate CF links.

Author: done

1. To Export logs, refer to [Manually subscribe AWS Kinesis Firehose stream to an existing CloudWatch Log Group](https://help.sumologic.com/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/#manually-subscribeaws-kinesis-firehose-stream-to-an-existing-cloudwatch-log-group).

> 1. Enable access logging with JSON as Log format for each APIs:

Reviewer 3: need to enable logging first and then export

>    1. [REST APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/apigateway-logging-to-kinesis.html)

Reviewer 3: Need to specify what format customers need to specify while configuring.

Author: done

Reviewer 3: We need to specify exactly the json fields you remember the json keys and values which we are using in our app.

Reviewer 2: @shivani-sumo Looks like this specific steps requested by Himanshu is not added in the doc. Line https://github.com/SumoLogic/sumologic-documentation/pull/182. Can you please check?

Author: It's there at L#265

   1. [HTTP APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/http-api-logging.html)
   1. [WebSocket APIs](https://docs.aws.amazon.com/apigateway/latest/developerguide/websocket-api-logging.html)

### Enable enhanced metrics

Enable enhanced metrics for REST APIs:

> 1. Enable access logs in **Logs and tracing** section. 

Reviewer 3: WE may have to add instructions from the beginning
1. Sign in to the API Gateway console at https://console.aws.amazon.com/apigateway.

2. Do one of the following:

    Choose an existing API, and then choose a stage.
    Create an API and deploy it to a stage.

3. In the main navigation pane, choose Stages.

Reviewer 2: @shivani-sumo Looks like this specific steps requested by Himanshu is not added in the doc. Line #182.

Author: Those steps are already there, written in different way

| fields apiName, namespace, apiId
```

> Create a field extraction rule for cloudwatch logs:

Reviewer 3: Create/Update Field Extraction Rule(s) for CloudWatch logs

#### Enable enhanced metrics

API Gateway will not send these metrics unless you have explicitly enabled detailed CloudWatch metrics. Enabling such metrics will incur additional charges to your account. For pricing information, see [Amazon CloudWatch Pricing](https://aws.amazon.com/cloudwatch/pricing/). Follow the below instructions for your respective API types.

##### Enable enhanced metrics for REST APIs:

1. Open the [Amazon API Gateway console](https://aws.amazon.com/api-gateway), then click on your REST API.
1. Select stage, then navigate to the **Logs and tracing** section.
1. Click **Edit**.
1. Select the **Detailed metrics** checkbox.
1. Click **Save changes**.

##### Enable enhanced metrics for HTTPS APIs:

1. Open the [Amazon API Gateway console](https://aws.amazon.com/api-gateway/), and click on your HTTPS API.
1. Under the **Monitor** section, click **Metrics**.
1. Select the stage, and click **Edit**.
1. Toggle the **enabled** button to enable the **Detailed route metrics**.
1. Click **Save**.

> 1. Click **Deploy**.<br/><img src={useBaseUrl('img/integrations/amazon-aws/HTTP_API_Enhanced_Metrics.png')} alt="AWS API Gateway" />

Reviewer: ```suggestion
1. Click **Deploy**.<br/><img src={useBaseUrl('img/integrations/amazon-aws/HTTP_API_Enhanced_Metrics.png')} alt="AWS API Gateway" style={{border: '1px solid gray'}} width="800" />

 />
```

##### Enable enhanced metrics for WebSocket APIs:

> 1. You can do this by calling the [UpdateStage](https://docs.aws.amazon.com/apigatewayv2/latest/api-reference/apis-apiid-stages-stagename.html) action of the API Gateway V2 REST API to update the `detailedMetricsEnabled` property to `true`. Alternatively, you can call the [update-stage](https://docs.aws.amazon.com/cli/latest/reference/apigatewayv2/update-stage.html) AWS CLI command to update the DetailedMetricsEnabled property to true as shown below:
> 1. Run the following command in your terminal:

Reviewer: ```suggestion
You can do this by calling the [UpdateStage](https://docs.aws.amazon.com/apigatewayv2/latest/api-reference/apis-apiid-stages-stagename.html) action of the API Gateway V2 REST API to update the `detailedMetricsEnabled` property to `true`. 

Alternatively, you can call the [update-stage](https://docs.aws.amazon.com/cli/latest/reference/apigatewayv2/update-stage.html) AWS CLI command to update the DetailedMetricsEnabled property to true. Run the following command in your terminal:
```

   ```sh
   aws apigatewayv2 update-stage --api-id <API_ID> --stage-name <STAGE_NAME> --default-route-settings <YOUR_ROUTE_SETTINGS> --output <OUTPUT_FORMAT> --region <REGION>
   ```

   ```sh title="Example"
   aws apigatewayv2 update-stage --api-id 9pk1qlmpci --stage-name $default --default-route-settings "{\"DetailedMetricsEnabled\":true}" --output json --region eu-north-1
   ```

> ### Collect Access Logs for AWS API Gateway   

Reviewer: ```suggestion
### Collect access logs for AWS API Gateway   
```

1. To your Hosted Collector, add an [AWS Kinesis Firehose for Logs Source](/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/).
   1. **Name**. Enter a name to display the new Source.
   2. **Description**. Enter an optional description.
   3. **Enable S3 Replay**. Do not check this option.
   4. **Source Category**. Enter `aws/apigateway`.
   5. **Fields**. Add below fields in it:
      1. Add an **account** field and assign it a value that is a friendly name/alias to your AWS account from which you are collecting logs. This name will appear in the Sumo Logic Explorer View. Logs can be queried via the `account field`.

>       2. Add **region**, and **accountid** fields and assign it's respective values.

Reviewer: ```suggestion
      2. Add **region** and **accountid** fields and assign their respective values.
```

   6. **Enable Timestamp Parsing**. Select the **Extract timestamp information from log file entries** check box.
   7. **Time Zone**. Select **Use time zone from log file. If none is detected use**, and select **Use Collector Default** from the dropdown.
   8. **Timestamp Format.** Select **Automatically detect the format**.
   9. **Enable Multiline Processing**. Select the **Detect messages spanning multiple lines** check box, and select **Infer Boundaries**.
   10. Click **Save**.
   11. Save the given URL of the source for next step.
2. [Create Stack](/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/#cloudformation-template) in AWS console with given CloudFormation Template.

> 3. Enable access logging with below exact JSON Log formats for each APIs:

Reviewer: ```suggestion
3. Enable access logging with the following exact JSON log formats for each API:
```

     "identityClientCertValidityNotBefore": "$context.identity.clientCert.validity.notBefore",
     "identityClientCertValidityNotAfter": "$context.identity.clientCert.validity.notAfter",
     "identityUserAgent": "$context.identity.userAgent",
     "identityUserArn": "$context.identity.userArn",
     "integrationError": "$context.integration.error",
     "integrationIntegrationStatus": "$context.integration.integrationStatus",
     "integrationLatency": "$context.integration.latency",
     "integrationRequestId": "$context.integration.requestId",
     "integrationStatus": "$context.integration.status",
     "contextIntegrationLatency": "$context.integrationLatency",
     "contextIntegrationStatus": "$context.integrationStatus",
     "path": "$context.path",
     "requestTimeEpoch": "$context.requestTimeEpoch",
     "stage": "$context.stage"
   }
   ```
4. To Export logs, refer to [Manually subscribe AWS Kinesis Firehose stream to an existing CloudWatch Log Group](/docs/send-data/hosted-collectors/amazon-aws/aws-kinesis-firehose-logs-source/#manually-subscribeaws-kinesis-firehose-stream-to-an-existing-cloudwatch-log-group).

### Collect AWS API Gateway CloudTrail Logs

> To your Hosted Collector, add an [AWS CloudTrail Source](/docs/send-data/hosted-collectors/amazon-aws/aws-cloudtrail-source.md) using the below instructions.

Reviewer: ```suggestion
To your Hosted Collector, add an [AWS CloudTrail Source](/docs/send-data/hosted-collectors/amazon-aws/aws-cloudtrail-source.md) using the instructions below.
```

> #### Collect CloudTrail Lambda Data Events

Reviewer: ```suggestion
#### Collect CloudTrail Lambda data events
```

To configure a CloudTrail Source, perform these steps:

1. [Grant Sumo Logic access](/docs/send-data/hosted-collectors/amazon-aws/grant-access-aws-product) to an Amazon S3 bucket.
2. [Create a trail for your AWS account](https://docs.aws.amazon.com/awscloudtrail/latest/userguide/cloudtrail-create-and-update-a-trail.html) and for more information on what events are logged refer [API gateway API calls documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/cloudtrail.html).
3. Confirm that logs are being delivered to the Amazon S3 bucket.
4. Add an [AWS CloudTrail Source](/docs/send-data/hosted-collectors/amazon-aws/aws-cloudtrail-source.md) to Sumo Logic.
5. While configuring the CloudTrail log source, the following field can be added to the source:
    * Add an **account** field and assign it a value that is a friendly name/alias to your AWS account from which you are collecting logs. This name will appear in the Sumo Logic Explorer View. Logs can be queried via the “account field”.

> <img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/AWS-Lambda/Fields.png')} alt="Fields" />

Reviewer: ```suggestion
<img src={useBaseUrl('https://sumologic-app-data-v2.s3.amazonaws.com/dashboards/AWS-Lambda/Fields.png')} alt="Fields" style={{border: '1px solid gray'}} width="400" />
```

> ### Centralized AWS CloudTrail log Collection

Reviewer: ```suggestion
### Centralized AWS CloudTrail log collection
```

> In case you have a centralized collection of CloudTrail logs and are ingesting them from all accounts into a single Sumo Logic CloudTrail log source, create the following Field Extraction Rule to map proper AWS account(s) friendly name/alias. You'll need to create it if not already present or update it as required.

Reviewer: ```suggestion
In case you have a centralized collection of CloudTrail logs and are ingesting them from all accounts into a single Sumo Logic CloudTrail log source, create the following field extraction rule to map proper AWS account(s) friendly name/alias. You'll need to create it if not already present or update it as required.
```

:::

For **Metadata**, add an **account** field to the source and assign it a value that is a friendly name/alias to your AWS account from which you are collecting metrics. This name will appear in the Sumo Logic Explorer View. Metrics can be queried via the “account field”.
#### Enable enhanced metrics

> API Gateway will not send these metrics unless you have explicitly enabled detailed CloudWatch metrics. Enabling such metrics will incur additional charges to your account. For pricing information, see Amazon CloudWatch Pricing.
> Follow below instruction for your respective api types.

Reviewer 4: ```suggestion
API Gateway will not send these metrics unless you have explicitly enabled detailed CloudWatch metrics. Enabling such metrics will incur additional charges to your account. For pricing information, see [Amazon CloudWatch Pricing](https://aws.amazon.com/cloudwatch/pricing/). Follow the below instructions for your respective API types.
```

##### Enable enhanced metrics for REST APIs:

> 1. Open the API [Gateway console](https://aws.amazon.com/api-gateway/), and click on your REST API.

Reviewer 4: ```suggestion
1. Open the [API Gateway console](https://aws.amazon.com/api-gateway/), then click on your REST API.
```

> 1. Open the API [Gateway console](https://aws.amazon.com/api-gateway/), and click on your REST API.
> 1. Select stage, and navigate to **Logs and tracing** section. 
> 1. Click on **Edit**. 
> 1. Select **Detailed metrics** checkbox.

Reviewer 4: ```suggestion
1. Open the [Amazon API Gateway console](https://aws.amazon.com/api-gateway), then click on your REST API.
1. Select stage, then navigate to the **Logs and tracing** section. 
1. Click **Edit**. 
1. Select the **Detailed metrics** checkbox.
```

1. Click **Save changes**.

##### Enable enhanced metrics for HTTPS APIs:

> 1. Open the API [Gateway console](https://aws.amazon.com/api-gateway/), and click on your HTTPS API.
> 1. Go to **Metrics** under **Monitor**. 
> 1. Select the stage, and click on **Edit**.
> 1. Toggle the **enabled** button to enable the **Detailed route metrics**. 
> 1. Click **Save**.
> 1. Click on **Deploy**.<br/><img src={useBaseUrl('img/integrations/amazon-aws/HTTP_API_Enhanced_Metrics.png')} alt="AWS API Gateway" />

Reviewer 4: ```suggestion
1. Open the API [Gateway console](https://aws.amazon.com/api-gateway/), and click on your HTTPS API.
1. Under the**Monitor** section, click **Metrics**.
1. Select the stage, and click **Edit**.
1. Toggle the **enabled** button to enable the **Detailed route metrics**. 
1. Click **Save**.
1. Click **Deploy**.<br/><img src={useBaseUrl('img/integrations/amazon-aws/HTTP_API_Enhanced_Metrics.png')} alt="AWS API Gateway" />
```

      "identityClientCertSubjectDN": "$context.identity.clientCert.subjectDN",
      "identityClientCertIssuerDN": "$context.identity.clientCert.issuerDN",
      "identityClientCertSerialNumber": "$context.identity.clientCert.serialNumber",
      "identityClientCertValidityNotBefore": "$context.identity.clientCert.validity.notBefore",
      "identityClientCertValidityNotAfter": "$context.identity.clientCert.validity.notAfter",
      "identityUserAgent": "$context.identity.userAgent",
      "identityUserArn": "$context.identity.userArn",
      "integrationError": "$context.integration.error",
      "integrationIntegrationStatus": "$context.integration.integrationStatus",
      "integrationLatency": "$context.integration.latency",
      "integrationRequestId": "$context.integration.requestId",
      "integrationStatus": "$context.integration.status",
      "contextIntegrationLatency": "$context.integrationLatency",
      "contextIntegrationStatus": "$context.integrationStatus",
      "path": "$context.path",
      "requestTimeEpoch": "$context.requestTimeEpoch",
      "stage": "$context.stage"
   }
   ```

>    * Enable Access logs for WebSocket APIs by referring to [AWS documentation](https://docs.aws.amazon.com/apigateway/latest/developerguide/set-up-logging.html#set-up-access-logging-using-console) and  when you specify the `Log format` field use the below JSON.

Reviewer 3: screenshot missing for all 3

### Overview

The **AWS API Gateway - Overview** dashboards provides insights into API Gateway performance throughout your infrastructure, including API calls, latency, client and server-side errors, connect and message count, data processed, API cache hits, and back-end cache misses.

There are three Overview dashboards: **AWS API Gateway - Overview (REST API)**, **AWS API Gateway - Overview (HTTP API)**, and **AWS API Gateway - Overview (WebSocket API)**.

To collect `CacheHitCount` and `CacheMissCount` metrics, enable your API cache:
* On the **AWS console**, Go to **API Gateway Service** and select Specific API -> **Stages**.
* Select Specific **Stage** and go to the **Settings** tab.
* The API should have the "**Enable API Cache**" checkbox enabled to enable API caching and to collect the CacheHitCount and CacheMissCount metrics.

> Use this dashboards to:

Reviewer: ```suggestion
Use these dashboards to:
```

* Monitor successful and failed API Gateway events, users and user agents / fail activities, and failure reasons
* Monitor requests coming in from known malicious IP addresses detected via [Sumo Logic Threat Intel](/docs/integrations/security-threat-detection/threat-intel-quick-analysis#03_Threat-Intel-FAQ)

<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Audit-Events.png')} alt="Audit Events" />

### Latency and Cache

The **AWS API Gateway - Latency, Cache** dashboards provides insights into API Gateway performance including API requests, latency, integration latency, and its statistics, API cache hits, and back-end cache misses.

There are two dashboards related to Latency and Cache: **AWS API Gateway - Latency, Cache (REST API)** and **AWS API Gateway - Latency (HTTP and WebSocket API)**.

> Use this dashboards to:

Reviewer: ```suggestion
Use these dashboards to:
```

* Monitor the responsiveness of the backend (integration latency), comparing times (in milliseconds) between API Gateway relay requests to and receiving a response back from the backend.
* Monitor API cache hits and misses to optimize cache capacities across your infrastructure and achieve desired performance.
* Compare API requests made today, yesterday, and last week to identify any abnormal deviations.

#### AWS API Gateway - Latency, Cache (REST API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Latency-Cache.png')} alt="Latency, Cache (REST API)" />

#### AWS API Gateway - Latency (HTTP and WebSocket API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Latency-Cache.png')} alt="Latency (HTTP and WebSocket API)" />

### Errors

The **AWS API Gateway - Errors** dashboards provides insights into API Gateway HTTP 4xx and 5xx code errors throughout your infrastructure, including API requests, client errors, integration errors, execution errors, client-side errors, and server-side errors.

> There are two dashboards related to Errors: **AWS API Gateway - Errors (REST API)** and **AWS API Gateway - Errors (HTTP and WebSocket API)**.

Reviewer: ```suggestion
There are two dashboards related to errors: **AWS API Gateway - Errors (REST API)** and **AWS API Gateway - Errors (HTTP and WebSocket API)**.
```

> Use this dashboards to:

Reviewer: ```suggestion
Use these dashboards to:
```

* Monitor the total number of client-side errors based on API name and region across your infrastructure.
* Monitor the total number of server-side errors based on API name and region across your infrastructure.
* Monitor the total number of client, integration and execution errors based on API name and region across your infrastructure.
* Compare API requests made today, yesterday, and last week to identify any abnormal deviations.

#### AWS API Gateway - Errors (REST API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-4XX-and-5XX-Errors.png')} alt="Errors (REST API)" />

#### AWS API Gateway - Errors (HTTP and WebSocket API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-4XX-and-5XX-Errors.png')} alt="Errors (HTTP and WebSocket API)" />

### Enhanced Monitoring

> The **AWS API Gateway - Enhanced Monitoring** dashboards provides detailed insights into API Gateway performance throughout your infrastructure, including the number and types of API calls, API resources, cache hits, and misses, latency averages, data proccessed, and errors by HTTP method, resource and stage.

Reviewer 4: ```suggestion
The **AWS API Gateway - Enhanced Monitoring** dashboards provides detailed insights into API Gateway performance throughout your infrastructure, including the number and types of API calls, API resources, cache hits, and misses, latency averages, data processed, and errors by HTTP method, resource and stage.
```

For your API Gateway instance to send enhanced metrics, you must have explicitly enabled detailed CloudWatch metrics. You can do this in the AWS management console under a Stage settings tab by selecting Enable CloudWatch Metrics. Alternatively, you can call the update-stage AWS CLI command to update the metrics enabled property to True.

There are three dashboards related to Enhanced Monitoring: **AWS API Gateway - Enhanced Monitoring (REST API)**, **AWS API Gateway - Enhanced Monitoring (HTTP API)**, and **AWS API Gateway - Enhanced Monitoring (WebSocket API)**.

> Use this dashboards to:

Reviewer: ```suggestion
Use these dashboards to:
```

* Monitor API cache hits and misses to optimize cache capacities to achieve desired performance.
* Monitor the overall responsiveness of API calls and the backend origin servers
* Monitor the total number of client-side and server-side errors based on HTTP methods

#### AWS API Gateway - Enhanced Monitoring (REST API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Enhanced-Monitoring.png')} alt="Enhanced Monitoring (REST API)" />

#### AWS API Gateway - Enhanced Monitoring (HTTP API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Enhanced-Monitoring.png')} alt="Enhanced Monitoring (HTTP API)" />

#### AWS API Gateway - Enhanced Monitoring (WebSocket API)
<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Enhanced-Monitoring.png')} alt="Enhanced Monitoring (WebSocket API)" />

### AWS API Gateway - Access Logs - Overview

The **AWS API Gateway - Access Logs - Overview** dashboard provides insights on Request's latency, Request trend, Distribution of requests by Method, Stage, and Protocol, Client's location, Request status code trend, and slowest requests.

> Use this dashboards to:

Reviewer: ```suggestion
Use this dashboard to:
```

* Monitor Latency across your infrastructure for all kind of APIs.

> * Monitor total API call, average response size and length and request trend.

Reviewer: ```suggestion
* Monitor total API call, average response size, and length and request trend.
```

* Monitor API requests trend by stage, HTTP method, user agent and protocol.

<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Enhanced-Monitoring.png')} alt="Access Logs - Overview" />

### AWS API Gateway - Access Logs - Details

> The **AWS API Gateway - Access Logs - Details** dashboard provides insights on Request's latencies, Request trend, Distribution of requests by Method, Stage, and Protocol, Client's location, Request status code trend, and slowest requests

Reviewer: ```suggestion
The **AWS API Gateway - Access Logs - Details** dashboard provides insights on the request's latencies; request trends; distribution of requests by method, stage, and protocol; client's location, request status code trend, and slowest requests.
```

> Use this dashboards to get detailed distribution and trend of API requests across all type of APIs.

Reviewer: ```suggestion
Use this dashboard to get detailed distribution and trend of API requests across all type of APIs.
```

<img src={useBaseUrl('img/integrations/amazon-aws/AWS-API-Gateway-Enhanced-Monitoring.png')} alt="Access Logs - Details" />

### AWS API Gateway - Access logs - Errors

The **AWS API Gateway - Access logs - Errors** dashboard provide insights on statistics of Top 20 failed requests, Error messages trend, Client's location, Errors by response type, Recent authorizer errors, missing authentication token errors, and WAF errors

> Use this dashboards to:

Reviewer: ```suggestion
Use this dashboard to:
```

> * Monitor all the failed requests, error messages and it's trend.
> * Monitor failed requests with it's distribution by response type, client IP.

Reviewer: ```suggestion
* Monitor all the failed requests, error messages, and their trends.
* Monitor failed requests with their distribution by response type and client IP.
```
